### PR TITLE
Build windows

### PR DESCRIPTION
**Pull request by andrewsavage1**
b/328808661

### CODE REVIEW

**Comment by jellefoks** [comment]
The linux-modular build fails with this PR, e.g. 

```
./../base/power_monitor/battery_level_provider_unittest.cc:79:8: error: expected '}'
       BatteryDetails({true,
       ^
```

File: base/logging.cc [hunk 1]
```diff
@@ -1301,13 +1301,21 @@ ScopedLoggingSettings::ScopedLoggingSettings()
 ScopedLoggingSettings::~ScopedLoggingSettings() {
   // Re-initialize logging via the normal path. This will clean up old file
   // name and handle state, including re-initializing the VLOG internal state.
+#if defined(STARBOARD)
```
Comment (andrewsavage1) [inline base/logging.cc:1304]: Investigate CHECK macros and why they don't expand correctly. Fine to follow up with this

File: base/export_template.h [hunk 2]
```diff
@@ -136,9 +136,9 @@
 #define EXPORT_TEMPLATE_TEST_EXPORT_DLLEXPORT_EXPORT_DLLEXPORT(...) true
 
 EXPORT_TEMPLATE_TEST(DEFAULT, );
-EXPORT_TEMPLATE_TEST(DEFAULT, __attribute__((visibility("default"))));
-EXPORT_TEMPLATE_TEST(EXPORT_DLLEXPORT, __declspec(dllexport));
-EXPORT_TEMPLATE_TEST(DEFAULT, __declspec(dllimport));
+// EXPORT_TEMPLATE_TEST(DEFAULT, __attribute__((visibility("default"))));
```
Comment (andrewsavage1) [inline base/export_template.h:139]: These expand incorrectly on windows, follow up on this

File: net/base/ip_endpoint.h [hunk 3]
```diff
@@ -30,7 +30,7 @@
 // Similarly, just pull in the minimal header necessary on non-Windows platforms
 // to help with build performance.
 struct sockaddr;
-#if BUILDFLAG(IS_WIN)
+#if BUILDFLAG(IS_WIN) || defined(USE_HACKY_COBALT_CHANGES) && defined(COMPILER_MSVC)
```
Comment (andrewsavage1) [inline net/base/ip_endpoint.h:33]: See if after Garo's change we can just use sys/socket.h

File: third_party/abseil-cpp/absl/random/internal/BUILD.gn
Comment (andrewsavage1) [inline third_party/abseil-cpp/absl/random/internal/BUILD.gn:1]: remove formatting on this file

File: third_party/googletest/src/googlemock/include/gmock/gmock-nice-strict.h [hunk 4]
```diff
@@ -143,8 +143,10 @@ class GTEST_INTERNAL_EMPTY_BASE_CLASS NiceMock
                 "https://google.github.io/googletest/"
                 "gmock_cook_book.html#NiceStrictNaggy");
   NiceMock() : MockClass() {
+#if !defined(COMPILER_MSVC)
```
Comment (andrewsavage1) [inline third_party/googletest/src/googlemock/include/gmock/gmock-nice-strict.h:146]: Add a TODO for this

File: third_party/icu/BUILD.gn [hunk 5]
```diff
@@ -80,10 +80,12 @@ config("icu_config") {
     defines += [ "ICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_STATIC" ]
   }
 
+  if (!is_starboard) {
```
Comment (andrewsavage1) [inline third_party/icu/BUILD.gn:83]: Add a TODO to investigate this

File: net/third_party/quiche/overrides/quiche_platform_impl/quiche_default_proof_providers_impl.cc [hunk 6]
```diff
@@ -87,15 +87,16 @@ std::unique_ptr<quic::ProofSource> CreateDefaultProofSourceImpl() {
   auto proof_source = std::make_unique<net::ProofSourceChromium>();
   proof_source->SetTicketCrypter(std::make_unique<quic::SimpleTicketCrypter>(
       quic::QuicChromiumClock::GetInstance()));
-  CHECK(proof_source->Initialize(
+  auto result = proof_source->Initialize(
```
Comment (andrewsavage1) [inline net/third_party/quiche/overrides/quiche_platform_impl/quiche_default_proof_providers_impl.cc:90]: Fix CHECK macro

File: net/dns/dns_names_util_unittest.cc [hunk 7]
```diff
@@ -773,7 +773,7 @@ TEST(DnsUtilTest, CanonicalizeNames) {
   EXPECT_EQ(UrlCanonicalizeNameIfAble("g{oo}gle.test"), "g%7Boo%7Dgle.test");
   EXPECT_EQ(UrlCanonicalizeNameIfAble("G{OO}GLE.test"), "g%7Boo%7Dgle.test");
 
-  // g�gle.test
+  // g�gle.test
```
Comment (andrewsavage1) [inline net/dns/dns_names_util_unittest.cc:776]: Revert

File: net/dns/address_info.cc [hunk 8]
```diff
@@ -192,7 +192,7 @@ std::unique_ptr<addrinfo, FreeAddrInfoFunc> AddrInfoGetter::getaddrinfo(
 #if BUILDFLAG(IS_ANDROID)
     *out_os_error = android::GetAddrInfoForNetwork(network, host.c_str(),
                                                    nullptr, hints, &ai);
-#elif BUILDFLAG(IS_WIN)
+#elif BUILDFLAG(IS_WIN) || defined(COMPILER_MSVC)
     *out_os_error = WSAEOPNOTSUPP;
     return rv;
 #else
```
Comment (andrewsavage1) [inline net/dns/address_info.cc:198]: Really we should be able to use the POSIX types

File: base/metrics/persistent_memory_allocator.cc [hunk 9]
```diff
@@ -1176,12 +1176,14 @@ void FilePersistentMemoryAllocator::FlushPartial(size_t length, bool sync) {
...
   DCHECK_NE(EINVAL, result);
-#elif BUILDFLAG(IS_POSIX) || BUILDFLAG(IS_FUCHSIA) || defined(STARBOARD)
+#elif BUILDFLAG(IS_POSIX) || BUILDFLAG(IS_FUCHSIA)
   // On POSIX, "invalidate" forces _other_ processes to recognize what has
   // been written to disk and so is applicable to "flush".
   int result = ::msync(const_cast<void*>(data()), length,
                        MS_INVALIDATE | (sync ? MS_SYNC : MS_ASYNC));
   DCHECK_NE(EINVAL, result);
+#elif defined(STARBOARD)
+  msync(const_cast<void*>(data()), length, 0);
```
Comment (andrewsavage1) [inline base/metrics/persistent_memory_allocator.cc:1186]: The last flag for msync here is wrong

File: net/disk_cache/disk_cache_backend.cc [hunk 10]
```diff
@@ -17,12 +22,293 @@
...
+ public:
+  std::unique_ptr<disk_cache::BackendFileOperations> Bind(
+      scoped_refptr<base::SequencedTaskRunner> task_runner) override {
+    return std::make_unique<disk_cache::TrivialFileOperations>();
+  }
+};
+
+}
+
+BackendResult::BackendResult() = default;
```
Comment (jellefoks) [inline net/disk_cache/disk_cache_backend.cc:65]: Please add a note here that this is copied from net/disk_cache/disk_cache.cc with a hint why it's copied.

File: net/cert/cert_verify_proc_builtin.cc [hunk 11]
```diff
@@ -123,6 +123,19 @@ base::Value::Dict NetLogPathBuilderResult(
   return dict;
 }
 
+#if defined(STARBOARD)
+RevocationPolicy DefaultRevocationPolicy() {
```
Comment (jellefoks) [inline net/cert/cert_verify_proc_builtin.cc:127]: I think this can instead be the constructor for `RevocationPolicy`

**Review by jellefoks — APPROVED** [review]
My comments here are only nits.

File: base/metrics/histogram_macros.h [hunk 12]
```diff
@@ -76,11 +76,12 @@
 // example). For scoped enums, this is awkward since it requires casting the
 // enum to an arithmetic type and adding one. Instead, prefer the two argument
 // version of the macro which automatically deduces the boundary from kMaxValue.
+#define CR_EXPAND_ARG(arg) arg
```
[the remark this answers is not in the document]
Comment (andrewsavage1) [reply base/metrics/histogram_macros.h:79]: We can't, because the macro is evaluated when macro below is used, not in the definition of the macro below

**Pull request merged**